src/frontend/js/components/SaleTunnel/CreditCardSelector/index.spec.tsx

```diff
@@ -0,0 +1,275 @@
+import { screen, within } from '@testing-library/react';
+import { useMemo, useState } from 'react';
+import fetchMock from 'fetch-mock';
+import { faker } from '@faker-js/faker';
+import userEvent from '@testing-library/user-event';
+import { RichieContextFactory as mockRichieContextFactory } from 'utils/test/factories/richie';
+import { CreditCardSelector } from 'components/SaleTunnel/CreditCardSelector/index';
+import { render } from 'utils/test/render';
+import { CreditCard } from 'types/Joanie';
+import {
+  CredentialOrderFactory,
+  CreditCardFactory,
+  ProductFactory,
+} from 'utils/test/factories/joanie';
+import { SaleTunnelProps } from 'components/SaleTunnel/index';
+import { setupJoanieSession } from 'utils/test/wrappers/JoanieAppWrapper';
+import { expectNoSpinner, expectSpinner } from 'utils/test/expectSpinner';
+import { SaleTunnelStep, SaleTunnelContext, SaleTunnelContextType } from '../GenericSaleTunnel';
+
+jest.mock('utils/context', () => ({
+  __esModule: true,
+  default: mockRichieContextFactory({
+    authentication: { backend: 'fonzie', endpoint: 'https://auth.test' },
+    joanie_backend: { endpoint: 'https://joanie.endpoint' },
+  }).one(),
+}));
+
+describe('CreditCardSelector', () => {
+  setupJoanieSession();
+
+  const buildWrapper = () => {
+    const contextRef = {
+      current: {} as SaleTunnelContextType,
+    };
+
+    const Wrapper = () => {
+      const [creditCard, setCreditCard] = useState<CreditCard>();
+      const context: SaleTunnelContextType = useMemo(
+        () => ({
+          eventKey: 'eventKey',
+          order: CredentialOrderFactory().one(),
+          product: ProductFactory().one(),
+          props: {} as SaleTunnelProps,
+          setBillingAddress: jest.fn(),
+          creditCard,
+          setCreditCard,
+          onPaymentSuccess: jest.fn(),
+          step: SaleTunnelStep.PAYMENT,
+        }),
+        [creditCard],
+      );
+      contextRef.current = context;
+
+      return (
+        <SaleTunnelContext.Provider value={context}>
+          <CreditCardSelector />
+        </SaleTunnelContext.Provider>
+      );
+    };
```

Comment:
As I use the new `render` method this seems not needed, nope?

Reply:
My idea was here to replace `<CreditCardSelector />` by a `{children}` and add the default wrapper used by our custom `render`. `buildWrapper` would become a component, juste `Wrapper`
Then when we render it, we can replace default render's wrapper with our custom one using the option `{wrapper: Wrapper}`.

I Think both work the same. I juste try here to have wrapper component instead of build function.